projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -3075,10 +3081,13 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
         this.deleteRowFromData(rowId, index);
         this._pipeTrigger++;
         this.cdr.markForCheck();
-
+        const dataAfterDelete = !this.transactions.enabled ? data : this.dataWithAddedInTransactionRows;
         this.refreshSearch();
-        if (data.length % this.perPage === 0 && this.isLastPage && this.page !== 0) {
-            this.page--;
+        if (dataAfterDelete.length % this.perPage === 0 && this.isLastPage && this.page !== 0) {
```

Comment:
These checks and the one below could be simplified to something like 
```suggestion
        if (dataAfterDelete.length % this.perPage === 0 && dataAfterDelete.length / this.perPage - 1 < this.page) {
```
where `dataAfterDelete.length / this.perPage` is pretty much a recalculated `totalPages`.
Actually, when you have transactions, the `onStateChange` emitter will actually retrigger the pipes and `totalPages` is up to date at this point and you can probably use it